### PR TITLE
Add POC jsonviewer that works anywhere

### PR DESCRIPTION
**Pull request by bovine3dom**
1) doesn't work at the moment
2) should make this part of `tabopen` to handle data: URIs

### CODE REVIEW

File: src/excmds.ts [hunk 1]
```diff
@@ -4206,11 +4204,13 @@ export function viewconfig(...key: string[]) {
...
- * View a JSON object in Firefox's JSON viewer. NB: only works well from a "normal" page, e.g not a Tridactyl help or new tab page.
+ * View a JSON object in Firefox's JSON viewer.
  */
-//#content
-export function jsonview(...json: string[]) {
-    window.location.href = "data:application/json," + json.join(" ").replace(/#/g, "%23").replace(/ /g, "%20")
+//#background
+export async function jsonview(...json: string[]) {
+    const tab = await tabopen("-w")
+    const url = "data:application/json," + json.join(" ").replace(/#/g, "%23").replace(/ /g, "%20")
```
Comment (Rummskartoffel) [inline src/excmds.ts:4212]:
```suggestion
    const url = "data:application/json," + encodeURIComponent(json.join(" "))
```

File: src/excmds.ts [hunk 2]
```diff
@@ -4206,11 +4207,13 @@ export function viewconfig(...key: string[]) {
...
 /**
- * View a JSON object in Firefox's JSON viewer. NB: only works well from a "normal" page, e.g not a Tridactyl help or new tab page.
+ * View a JSON object in Firefox's JSON viewer.
  */
-//#content
-export function jsonview(...json: string[]) {
-    window.location.href = "data:application/json," + json.join(" ").replace(/#/g, "%23").replace(/ /g, "%20")
+//#background
+export async function jsonview(...json: string[]) {
+    const tab = await tabopen("-w")
```
Comment (bovine3dom) [inline src/excmds.ts:4214]: I just realised that this shouldn't work, right? `:viewconfig`'s whole problem was that it didn't work from the new tab page. We're now _always_ opening it from the new tab page.

I mean, I'm glad it does work, but I'm thinking that maybe `encodeURIComponent` was the real fix.

Comment (Rummskartoffel) [reply src/excmds.ts:4214]: Don't think it was. Try running `location = "data:application/json," + encodeURIComponent('{"foo": "bar"}')` in the new tab page's console. I'm still not getting the JSON viewer that way.

Comment (bovine3dom) [reply src/excmds.ts:4214]: so how on earth did we fix it? lol

Comment (Rummskartoffel) [reply src/excmds.ts:4214]: I'm guessing that the CSP bug that prevents the viewer from loading doesn't trigger through `tabs.executeScript()` for some reason. That's the only thing that's different, so it really can't be anything else. No idea *why* that is, though. Maybe we'll learn more once the bug I filed with Mozilla gets triaged.

**Pull request merged**